src/fitnesse/components/PluginsClassLoader.java

```diff
@@ -2,45 +2,54 @@
 
 import java.io.File;
 import java.lang.reflect.Method;
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
 
 
+/**
+ * Update the current thread class path with jars foundin a "plugins" directory.
+ */
 public class PluginsClassLoader {
 
-  private File pluginsDirectory;
+  public static ClassLoader loadPlugins(String rootPath) throws Exception {
+    File pluginsDirectory = new File(rootPath, "plugins");
 
-  public PluginsClassLoader(String rootPath) {
-    pluginsDirectory = new File(rootPath, "plugins");
+    URL[] urls = urlsForPlugins(pluginsDirectory);
+    if (urls.length > 0) {
+      return new URLClassLoader(urls, ClassLoader.getSystemClassLoader());
+    } else {
+      return ClassLoader.getSystemClassLoader();
+    }
   }
 
-  public void addPluginsToClassLoader() throws Exception {
-    if (pluginsDirectory.exists())
+  private static URL[] urlsForPlugins(File pluginsDirectory) throws Exception {
+    List<URL> urls = new ArrayList<>();
+
+    if (pluginsDirectory.exists() && pluginsDirectory.isDirectory())
       for (File plugin : pluginsDirectory.listFiles())
         if (plugin.getName().endsWith("jar"))
-          addItemsToClasspath(plugin.getCanonicalPath());
+          urls.addAll(toUrls(plugin.getCanonicalPath()));
```

Comment:
Each plugin file is a single element, right? So why not call `toUrl()`, and add that one url the list. The updating of the system property does not have to be done for each plugin found, but can be postponed to after full url list is assembled (and has items).

Reply:
Fixed.